src/aiida/orm/nodes/data/remote/stash/copy.py

```diff
@@ -0,0 +1,80 @@
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Data plugin that models a stashed folder on a remote computer."""
+
+from typing import List, Tuple, Union
+
+from aiida.common.datastructures import StashMode
+from aiida.common.lang import type_check
+from aiida.common.pydantic import MetadataField
+
+from .base import RemoteStashData
+
+__all__ = ('RemoteStashCopyData',)
+
+
+class RemoteStashCopyData(RemoteStashData):
+    """Data plugin that models a folder with files of a completed calculation job that has been stashed through a copy.
+
+    This data plugin can and should be used to stash files if and only if the stash mode is `StashMode.COPY`.
+    """
+
+    _storable = True
+
+    class Model(RemoteStashData.Model):
+        target_basepath: str = MetadataField(description='The the target basepath')
+        source_list: List[str] = MetadataField(description='The list of source files that were stashed')
+
+    def __init__(self, stash_mode: StashMode, target_basepath: str, source_list: List[str], **kwargs):
+        """Construct a new instance
+
+        :param stash_mode: the stashing mode with which the data was stashed on the remote.
+        :param target_basepath: the target basepath.
+        :param source_list: the list of source files.
+        """
+        super().__init__(stash_mode, **kwargs)
+        self.target_basepath = target_basepath
+        self.source_list = source_list
+
+        if stash_mode != StashMode.COPY:
```

Comment:
Seems a bit verbose but I really struggled with this when looking at this code after some time and wanted to change this again
```suggestion
        # We are consistent with the signature in `RemoteStashData` following LSP,
        # even though in this subclass we only support one stash mode which make this seem weirdly designed at first
        # glance, note there is also a subsclass for compressed options

        if stash_mode != StashMode.COPY:
```